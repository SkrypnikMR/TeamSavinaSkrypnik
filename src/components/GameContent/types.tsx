import { ITheme } from '../App/types';

export interface IDiv extends ITheme{
    background: string;
    margin: string;
    width: string;
    padding: string;
}
