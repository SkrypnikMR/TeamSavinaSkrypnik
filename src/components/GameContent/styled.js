import styled from 'styled-components';

export const StGameContent = styled.div`
background: #C4C4C4;
margin: 10px;
width: 70%;
padding: 10px;
`