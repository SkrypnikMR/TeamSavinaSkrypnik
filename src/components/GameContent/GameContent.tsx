import React from 'react';

import GameZone from '../GameZone';
import { useTheme } from '../Hook/useTheme';

import { StGameContent } from './styled';

const GameContent = () => {
    const { colors, theme } = useTheme();
    return (
        <StGameContent colors={colors} theme={theme}>
            Checkers
            <GameZone/>
        </StGameContent>
    );
};

export default GameContent;
