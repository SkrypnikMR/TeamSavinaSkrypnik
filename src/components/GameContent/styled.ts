import styled from 'styled-components';

import { IDiv } from './types';

export const StGameContent = styled.div < IDiv > `
    background: ${({ colors, theme }) => colors[theme].backgroundSecondary};
    margin: 10px;
    width: 70%;
    padding: 10px;
    border-radius: 3px;
    box-shadow: 0 2px 4px rgba(0, 0, 0, .2);
`;
