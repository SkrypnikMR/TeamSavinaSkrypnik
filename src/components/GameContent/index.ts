import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import GameContent from './GameContent';
import { getWinner } from '../../store/game/selectors';
import { cleanOldGame } from '../../store/game/actions';

const mapStateToProps = store => ({ winner: getWinner(store) });
const mapDispatchToProps = (dispatch: Dispatch) => ({ cleanOldGame: () => dispatch(cleanOldGame()) });

export default connect(mapStateToProps, mapDispatchToProps)(GameContent);
