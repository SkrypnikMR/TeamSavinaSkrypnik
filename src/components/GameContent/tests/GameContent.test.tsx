import React from 'react';
import { shallow, mount } from 'enzyme';

import GameContent from '../GameContent';

describe('GameContent', () => {
    it('Should match snapshot', () => {
        const component = shallow(<GameContent />);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render StGameContent', () => {
        const component = mount(<GameContent/>);
        expect(component.find('styled__StGameContent')).toHaveLength(1);
    });
    it('should render GameZone', () => {
        const component = mount(<GameContent/>);
        expect(component.find('GameZone')).toHaveLength(1);
    });
});
