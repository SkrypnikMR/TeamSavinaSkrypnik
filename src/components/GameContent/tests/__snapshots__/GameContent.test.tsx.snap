// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameContent Should match snapshot 1`] = `"<div class=\\"styled__StGameContent-sc-rfvg7b-0 iDkgGF\\">Checkers<div class=\\"styled__StGameZone-sc-1037dno-0 ftvCaM\\"><p>your_turn</p><div class=\\"styled__StTicTacToePlate-sc-ahqcne-0 bNLTCa\\"><div id=\\"0\\" class=\\"styled__StSingleSellTicTac-sc-1ruk552-0 QdPRA\\"></div><div id=\\"1\\" class=\\"styled__StSingleSellTicTac-sc-1ruk552-0 QdPRA\\"></div><div id=\\"2\\" class=\\"styled__StSingleSellTicTac-sc-1ruk552-0 QdPRA\\"></div><div id=\\"3\\" class=\\"styled__StSingleSellTicTac-sc-1ruk552-0 QdPRA\\"></div><div id=\\"4\\" class=\\"styled__StSingleSellTicTac-sc-1ruk552-0 QdPRA\\"></div><div id=\\"5\\" class=\\"styled__StSingleSellTicTac-sc-1ruk552-0 QdPRA\\"></div><div id=\\"6\\" class=\\"styled__StSingleSellTicTac-sc-1ruk552-0 QdPRA\\"></div><div id=\\"7\\" class=\\"styled__StSingleSellTicTac-sc-1ruk552-0 QdPRA\\"></div><div id=\\"8\\" class=\\"styled__StSingleSellTicTac-sc-1ruk552-0 QdPRA\\"></div></div></div></div>"`;
